src/Microsoft.Extensions.DependencyInjection/ServiceCollectionContainerBuilderExtensions.cs

```diff
@@ -29,7 +29,31 @@ public static IServiceProvider BuildServiceProvider(this IServiceCollection serv
         /// <returns>The<see cref="IServiceProvider"/>.</returns>
         public static IServiceProvider BuildServiceProvider(this IServiceCollection services, bool validateScopes)
         {
-            return new ServiceProvider(services, validateScopes);
+            return services.BuildServiceProvider(new ServiceProviderOptions { ValidateScopes = validateScopes });
+        }
+
+        /// <summary>
+        /// Creates an <see cref="IServiceProvider"/> containing services from the provided <see cref="IServiceCollection"/>
+        /// optionaly enabling scope validation.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> containing service descriptors.</param>
+        /// <param name="options">
+        /// Configures various service provider behaviors.
+        /// </param>
+        /// <returns>The<see cref="IServiceProvider"/>.</returns>
+        public static IServiceProvider BuildServiceProvider(this IServiceCollection services, ServiceProviderOptions options)
```

Comment:
Don't need to new up anything that's the only reason we use that pattern. It's also to avoid having to know the namespace the options type (but that's moot here)